libraries/AP_Mission/AP_Mission.h

```diff
@@ -291,6 +291,8 @@ class AP_Mission {
         _prev_nav_cmd_wp_index(AP_MISSION_CMD_INDEX_NONE),
         _last_change_time_ms(0)
     {
+        _singleton = this;
```

Comment:
Typical pattern is to add a HAL_SITL guarded check to this to ensure that `_singleton` was null when we created the object.